src/map/battle.cpp

```diff
@@ -3420,7 +3420,9 @@ static int battle_calc_equip_attack(struct block_list *src, int skill_id)
 		struct status_data *status = status_get_status_data(src);
 		map_session_data *sd = BL_CAST(BL_PC, src);
 
-		if (sd) // add arrow atk if using an applicable skill
+		// Cart Cannon is an arrow skill but adds arrow attack as mastery damage instead
```

Comment:
```suggestion
		// Add arrow atk if using an applicable skill
		// Cart Cannon is an exceptional arrow skill and adds arrow attack as mastery damage instead
```

Reply:
But you merged your check to the outer and not the inner (short) if.

Reply:
Reworked this part